src/firebolt/async_db/cursor.py

```diff
@@ -213,29 +257,52 @@ def _reset(self) -> None:
         self._descriptions = None
         self._rowcount = -1
         self._idx = 0
+        self._row_sets = []
+        self._next_set_idx = 0
 
     async def _do_execute_request(
         self,
         query: str,
-        parameters: Optional[Sequence[ParameterType]] = None,
+        parameters: Sequence[Sequence[ParameterType]],
         set_parameters: Optional[Dict] = None,
-    ) -> Response:
-        if parameters:
-            query = format_sql(query, parameters)
-
-        resp = await self._client.request(
-            url="/",
-            method="POST",
-            params={
-                "database": self.connection.database,
-                "output_format": JSON_OUTPUT_FORMAT,
-                **(set_parameters or dict()),
-            },
-            content=query,
-        )
+    ) -> None:
+        self._reset()
+        try:
+
+            queries = split_format_sql(query, parameters)
+
+            for query in queries:
+
+                start_time = time.time()
+                # our CREATE EXTERNAL TABLE queries currently require credentials,
+                # so we will skip logging those queries.
+                # https://docs.firebolt.io/sql-reference/commands/ddl-commands#create-external-table
+                if not re.search("aws_key_id|credentials", query, flags=re.IGNORECASE):
+                    logger.debug(f"Running query: {query}")
+
+                resp = await self._client.request(
+                    url="/",
+                    method="POST",
+                    params={
+                        "database": self.connection.database,
+                        "output_format": JSON_OUTPUT_FORMAT,
+                        **(set_parameters or dict()),
+                    },
+                    content=query,
+                )
+
+                await self._raise_if_error(resp)
+                self._append_query_data(resp)
+                logger.info(
+                    f"Query fetched {self.rowcount} rows in"
+                    f" {time.time() - start_time} seconds"
+                )
+
+            self._state = CursorState.DONE
 
-        await self._raise_if_error(resp)
-        return resp
+        except Exception:
+            self._state = CursorState.ERROR
+            raise
```

Comment:
Do we check anywhere for CursorState.ERROR? Or is this purely for user information?

Reply:
Yes, it's only for users